src/payment/payment.controller.ts

```diff
@@ -0,0 +1,27 @@
+import { Controller, Post, Body, Req, UseGuards } from '@nestjs/common';
+import { PaymentService } from './payment.service';
+import { CreatePaymentDto } from './dto/create-payment.dto';
+import { UserGuard } from '../common/guards/user.guard';
+import { ApiBearerAuth, ApiOperation, ApiResponse, ApiTags } from '@nestjs/swagger';
+
+@Controller('payment')
+@UseGuards(UserGuard)
+@ApiTags('payment')
+@ApiBearerAuth()
+export class PaymentController {
+  constructor(private readonly paymentService: PaymentService) {}
+
+  @ApiOperation({ summary: 'Handle payment' })
+  @ApiResponse({ status: 200, description: 'Payment processed successfully' })
+  @ApiResponse({ status: 400, description: 'Bad Request' })
+  @Post('')
+  async handlePayment(@Body() createPaymentDto: CreatePaymentDto, @Req() req: any) {
+    const { totalAmount, paymentOption } = createPaymentDto;
+
+    if (paymentOption === 'cash') {
```

Comment:
first of all, don't use unnecessary if else in JS. 
if (paymentOption === 'cash') {
   await this.paymentService.processCashPayment(req.user.id, totalAmount);
   return;
}
await this.paymentService.processPayment(req.user.id, totalAmount, paymentOption);

but it possible that we will use not only cash and card, but other payment option, better to create map with handlers
{
  cash: this.paymentService.processCashPayment(req.user.id, totalAmount);
  card: this.paymentService.processPayment(req.user.id, totalAmount, paymentOption);
  usdt: .....
  sofort: ....
}
add also validation of paymentOption it should be only 'cash' or 'card' for now otherwise you should throw the Bad Request error.